.github/workflows/ci-upstream.yml

```diff
@@ -0,0 +1,35 @@
+name: Upstream
+
+on:
+  schedule:
+    - cron: "0 1 * * *"
+  push:
+  pull_request:
```

Comment:
Currently these CI builds will run on pushes to forks. We should consider only running on pushes / pull requests to the main branch.